docs/posts/1tbc/index.qmd

```diff
@@ -0,0 +1,755 @@
+
+```{python}
+#| code-fold: true
+#| code-summary: "Show code to read and analyze the benchmark data"
+import ibis
+import gcsfs
+import plotly.express as px
+
+from great_tables import GT, md
+
+px.defaults.template = "plotly_dark"
+
+ibis.set_backend("duckdb")
+ibis.options.interactive = True
+ibis.options.repr.interactive.max_rows = 3
+
+fs = gcsfs.GCSFileSystem()
+ibis.get_backend().register_filesystem(fs)
+
+t = (
+    ibis.read_parquet(
+        "gs://ibis-bench/1tbc/cache/file_id=*.parquet",
+    )
+    .select(
+        "system",
+        "sf",
+        "n_partitions",
+        "query_number",
+        "execution_seconds",
+        "timestamp",
+    )
+    .mutate(timestamp=ibis._["timestamp"].cast("timestamp"))
+    .order_by("system", "query_number")
+    .cache()
+)
+
+systems = sorted(t.distinct(on="system")["system"].collect().to_pyarrow().as_py())
+
+agg = (
+    t.mutate(
+        run_num=ibis.row_number().over(
+            group_by=["system", "sf", "n_partitions", "query_number"],
+            order_by=["timestamp"],
+        )
+    )
+    .relocate(t.columns[:4], "run_num")
+    .group_by("system", "query_number", "run_num")
+    .agg(execution_seconds=ibis._["execution_seconds"].mean())
+    .order_by("system", "query_number", "run_num")
+)
+agg2 = (
+    agg.group_by("system", "query_number")
+    .agg(avg_execution_seconds=agg.execution_seconds.mean().round(2))
+    .order_by("system", "query_number")
+)
+piv = agg2.pivot_wider(
+    names_from="system", values_from=["avg_execution_seconds"]
+).order_by("query_number")
+
+
+def x_vs_y(piv, x, y):
+    return ibis.ifelse(
+        piv[x] < piv[y],
+        -1,
+        1,
+    ) * (
+        (
+            (piv[x] - piv[y])
+            / ibis.ifelse(
+                piv[y] > piv[x],
+                piv[x],
+                piv[y],
+            )
+        ).abs()
+    ).round(4)
+
+
+comparisons = [
+    ("ibis-datafusion", "ibis-duckdb"),
+    ("polars-lazy", "ibis-datafusion"),
+    ("polars-lazy", "ibis-duckdb"),
+]
+
+comparisons = {f"{x}_v_{y}": x_vs_y(piv, x, y) for x, y in comparisons}
+
+piv2 = piv.mutate(**comparisons)
+piv2 = piv2.order_by("query_number").relocate("query_number", systems)
+
+agg3 = (
+    agg2.group_by("system")
+    .agg(
+        queries_completed=agg2["avg_execution_seconds"].count(),
+        execution_seconds=agg2["avg_execution_seconds"].sum().round(2),
+        seconds_per_query=agg2["avg_execution_seconds"].mean().round(2),
+    )
+    .order_by(ibis.desc("queries_completed"))
+)
+agg3
+```
+
+`ibis-duckdb` completed all 22/22 queries **in under 30 minutes**. If you need
+to run batch data jobs on a similar amount of data, a laptop might be all you
+need!
+
+`ibis-datafusion` only completed 17/22 queries, though recall [3 are failing due
+to a bug that's already been
+fixed](../ibis-bench/index.qmd#failing-datafusion-queries). A new Python release
+for DataFusion hasn't been made yet, so we ran with the old version. Assuming
```

Comment:
I just updated this in https://github.com/ibis-project/ibis/pull/9506, so probably should re-run.

Reply:
I already ran with it (just `pip install --upgrade datafusion`, but the bug fix needed isn't in `39.0.0`. it was merged a bit after, so need the next release. I would like to have that but can do a follow-up

Reply:
Yep, sounds good!